cmd/kn/main.go

```diff
@@ -79,7 +81,12 @@ func run(args []string) error {
 			return err
 		}
 
-		return plugin.Execute(argsWithoutCommands(args, plugin.CommandParts()))
+		err = plugin.Execute(argsWithoutCommands(args, plugin.CommandParts()))
+		if err != nil {
+			// Used to not print `--kn help` message with plugin errors
+			pluginErr = true
```

Comment:
Can we please avoid to use global state variable to transport information ? And instead either extend the return signature to add a boolean indicating that the error is 'help-messagable' or return an error with a certain prefix.


This would be a simple solution but a more elegant solution would be to return a higher order, custom error type wich carries this information (and egg provides a method to be called like `err.HelpUsageMessage()` which can be empty so that it is not printed.

Reply:
Thanks, I added the `bool` to the return signature.